pkg/controllers/cstorvolumeconfig/volume_operations.go

```diff
@@ -312,6 +312,24 @@ func (c *CVCController) getOrCreateCStorVolumeResource(
 	return cvObj, err
 }
 
+func getPolicyBasedPoolList(
+	poolList *apis.CStorPoolInstanceList,
+	poolInfo []apis.ReplicaPoolInfo,
+) *apis.CStorPoolInstanceList {
+	policyBasedList := &apis.CStorPoolInstanceList{}
+	for _, info := range poolInfo {
+		for _, pool := range poolList.Items {
+			if pool.Name == info.PoolName {
+				policyBasedList.Items = append(
+					policyBasedList.Items,
+					pool,
+				)
```

Comment:
add `break` over here.

Reply:
If we add break here won't it return only one pool name in the list? We want all the pool names in the policy.

Reply:
Ok got it. Ignore my comment.